library/src/main/java/com/google/android/material/motion/runtime/Transaction.java

```diff
@@ -60,7 +60,10 @@ public void addPlan(Plan plan, Object target) {
   /**
    * Adds a named {@link Plan} to this Transaction, targeting the given object. The Plan overwrites any
    * previously added Plans with the same name.
+   * @deprecated  Named Plans should be added directly to the Scheduler instead of using Transactions. <br />
+   *              use {@link com.google.android.material.motion.runtime.Scheduler#addNamedPlan(NamedPlan, String, Object)} on the Scheduler instead
    */
+  @Deprecated
```

Comment:
nice! all public methods have been @deprecated. Can you @deprecate the class itself as well?

Also, every @deprecated tag should be followed by the version it was deprecated in. In your case, you should insert `3.0.0` since i'm about to push that release (this change will go live in 3.0.0)